docs/functions.json

```diff
@@ -341,23 +341,15 @@
     },
     {
         "signature": "extract(array, properties...)",
-        "summary": "Takes an array of objects and returns a new array by extracting named properties from each item.",
-        "detail": "If a single property is specified, the returned array is a flat array of values. If multiple properties\nare specified then each item is a dict of with those properties.",
+        "summary": "Takes a dict and returns a new dict by extracting only them named properties.",
+        "detail": "If a single property is specified, the function returns that single value. If multiple properties\nare specified the returned value is a new dict with those properties.",
```

Comment:
For the single case, doesn't that violate our rule about the type of functions being known? Or not because we know the # of parameters? May be good to call out behavior if a key isn't there (null)

Reply:
Do any of our migrations require this? If not, I would vote towards always returning a dict and punting this for now. (until we have the need)

Reply:
unfortunately yes (group names, attachment urls) - it's the returning a dict case that we don't currently need... `select` vs `extract` ? Or maybe compile time predictability is fine. i'm trying to convince myself of that right now.

Reply:
Hah. Maybe they should be more similar than different? 
`extract` and `extract_one`? 
`extract_dict` vs `extract`?

Reply:
I see single value being the common case so I'd lean toward `extract` vs `extract_dict`